### ISSUE TITLE
Action for generating coverage reports

### ISSUE THREAD

**Issue by garyttierney** [actions-rs/meta#1]
Is there any interest in getting support for an action that can generate code coverage reports? I have a simple thing [here](https://github.com/garyttierney/coverage-rs)[1] based on https://github.com/xd009642/tarpaulin, but it's limited in that it only supports x86 Linux.

I've also had some success with [grcov](https://github.com/mozilla/grcov) and rustc's profiling capabilities, which may work for other platforms.

I'd be willing to spend a bit of time getting this working.

[1]: example [here](https://github.com/garyttierney/secsp/runs/224360293) with [workflow](https://github.com/garyttierney/secsp/blob/chore/rowan-0.6/.github/workflows/ci.yml#L37-L49).

**Comment by lucasbrendel** [comment]
It would be nice to carry over the behavior of [cargo-travis](https://github.com/roblabla/cargo-travis) for code coverage as well as documentation publishing to gh-pages

**Comment by svartalf** [comment]
@garyttierney, yes, code coverage is definitely a thing to add! I had created #2 and #3 to track progress for `tarpaulin` and `grcov` actions respectively.

@lucasbrendel I'm strongly against re-implementing coverage reports uploading; there are official [coveralls](https://github.com/marketplace/actions/coveralls-github-action) and [codecov](https://github.com/marketplace/actions/codecov) Actions at the GH marketplace, it would be better to use them.
Action for doc publishing looks promising, though, here is a #4 :)

**Comment by lucasbrendel** [comment]
@svartalf i totally get not wanting to duplicate work. So would #2 and #3 cover creating coverage reports for something like codecov and then use the codecov action to upload?

Thanks for #4

**Comment by svartalf** [comment]
@lucasbrendel yep, that's exactly an idea: our Actions will generate the report file and you can do whatever you want with it later: upload to the coveralls/codecov, convert into the .html report with [genhtml](https://github.com/linux-test-project/lcov/blob/master/bin/genhtml) or anything else.
That approach is more flexible and will allow to build more complex workflows.

If you are interested, you can check the [`@actions-rs/grcov` "dev" branch](https://github.com/actions-rs/grcov/tree/dev) and [this example](https://github.com/heim-rs/heim/blob/gh-actions/.github/workflows/ci.yml#L85-L147), which gathers code coverage reports from the multiple parallel builds and uploads them to the coveralls service.
I should warn, though, that it is not even in "alpha" state yet, so things might break eventually, but at least it can show what I mean.
